vendor/github.com/darkowlzz/operator-toolkit/client/composite/client.go

```diff
@@ -0,0 +1,68 @@
+package composite
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+// Client is a composite client, composed of cached and unclient clients. It
+// can be used to query objects from cache and fall back to use the raw client
+// and get the object directly from the API server.
+type Client struct {
+	client.Client
+	Options
+
+	uncached client.Client
+}
+
+// ClientOption is used to configure the client.
+type Options struct {
+	// RawListing is used to perform raw listing operations, uncached.
+	RawListing bool
+}
+
+// NewClient creates and returns a composite Client.
+func NewClient(cached client.Client, uncached client.Client, opts Options) *Client {
+	return &Client{
+		Client:   cached,
+		Options:  opts,
+		uncached: uncached,
+	}
+}
+
+// NewClientFromManager combines a cached and an uncached client to return a
+// composite client. The default cache returned by the controller manager is a
+// cached client.
+func NewClientFromManager(mgr manager.Manager, opts Options) (client.Client, error) {
+	uncachedClient, err := client.New(mgr.GetConfig(), client.Options{Scheme: mgr.GetScheme()})
+	if err != nil {
+		return nil, err
+	}
+
+	return NewClient(mgr.GetClient(), uncachedClient, opts), nil
+}
+
+// Get first fetches the object using the cached client. If the object is not
+// found in the cached client, it retries using the uncached client.
+func (c *Client) Get(ctx context.Context, key client.ObjectKey, obj client.Object) error {
+	if cErr := c.Client.Get(ctx, key, obj); cErr != nil {
```

Comment:
q: why `cErr` and not just `err`?

Reply:
I think there were some other error variables before that I removed.
This is part of a change in style I've been practicing to avoid shadowing variables accidentally, especially error variables.